src/castTo/castTo.spec.ts

```diff
@@ -0,0 +1,26 @@
+import { castTo } from './castTo';
+
+describe('castTo function', () => {
+  const input = { a: 1, b: 2, c: 3 };
+  const array = [{ a: 1, b: 2, c: 3 }, { a: 4, b: 5, c: 6 }];
+
+  test('is a pure function', () => {
+    castTo({ a: x => x })(input);
+    expect(input).toMatchSnapshot();
+  });
+
+  test('{a: 1, b: 2, c:3} |> castTo === {a: 2, b: 2, c: 3}', () => {
```

Comment:
You miss the purpose of the function in the test.
Please add a test : `'{ a: '1', b: 2.1, c: 'A' } |> castTo(x => parseInt(x.a, 10), x => Math.floor(x.B), x => parseInt(x.c, 16)) === { a: 1, b: 2, c: 10 }'`

Reply:
Done